opendebates/forms.py

```diff
@@ -83,3 +86,41 @@ def save(self, commit=True):
 class OpenDebatesAuthenticationForm(AuthenticationForm):
     username = forms.CharField(max_length=254,
                                label="Username or Email")
+
+
+class MergeFlagForm(forms.ModelForm):
+    duplicate_of_url = forms.URLField()
+
+    class Meta:
+        model = Flag
+        fields = ('duplicate_of_url', )
+
+    def __init__(self, *args, **kwargs):
+        self.idea = kwargs.pop('idea')
+        self.voter = kwargs.pop('voter')
+        super(MergeFlagForm, self).__init__(*args, **kwargs)
+
+    def clean_duplicate_of_url(self):
+        # parse the URL and use Django's resolver to find the urlconf entry
+        path = urlparse(self.cleaned_data['duplicate_of_url']).path
+        url_match = resolve(path)
+
+        if url_match.url_name != 'vote':
+            raise forms.ValidationError('That is not the URL of a question.')
+
+        duplicate_of_pk = url_match.kwargs.get('id')
+        self.duplicate_of = Submission.objects.exclude(pk=self.idea.pk) \
```

Comment:
How about if we first check that the duplicate submission is not the same as the submission it's supposed to be a duplicate of, then we can give a more helpful error - like "Please enter the URL of the submission that this submission appears to be a duplicate of, not the URL of this submission".

We might also do separate checks for the duplicate submission existing, and then whether it's approved (though - do we expose the "approved" thing at all?  If not, then not.)